File: articles/motivation-goals-and-the-architecture-of-desire/r/analyze_motivation_goals_desire.R
root <- normalizePath(file.path(dirname(sys.frame(1)$ofile), ".."))
data_path <- file.path(root, "data", "synthetic_motivation_goals_desire.csv")
outputs <- file.path(root, "outputs")
dir.create(outputs, showWarnings = FALSE, recursive = TRUE)

data <- read.csv(data_path, stringsAsFactors = FALSE)

data$ownership_level <- ifelse(
  data$goal_ownership > median(data$goal_ownership),
  "higher_ownership",
  "lower_ownership"
)
data$conflict_level <- ifelse(
  data$goal_conflict > median(data$goal_conflict),
  "higher_conflict",
  "lower_conflict"
)
data$meaning_level <- ifelse(
  data$meaning_goal > median(data$meaning_goal),
  "higher_meaning",
  "lower_meaning"
)
data$motivational_profile <- paste(
  data$ownership_level,
  data$conflict_level,
  data$meaning_level,
  sep = "_"
)

data$high_conflict_low_ownership <- data$goal_conflict > median(data$goal_conflict) &
  data$goal_ownership < median(data$goal_ownership)
data$high_status_low_meaning <- data$status_goal > median(data$status_goal) &
  data$meaning_goal < median(data$meaning_goal)

context_summary <- aggregate(
  data[, c("approach_orientation", "status_orientation", "avoidance_security_orientation", "need_support", "goal_ownership", "motivational_quality", "goal_conflict", "persistence_score", "adaptive_disengagement", "life_direction_coherence", "well_being")],
  by = list(motivation_context = data$motivation_context),
  FUN = mean
)

profile_summary <- aggregate(
  data[, c("goal_ownership", "goal_conflict", "meaning_goal", "need_support", "motivational_quality", "life_direction_coherence", "persistence_score", "adaptive_disengagement", "well_being")],
  by = list(motivational_profile = data$motivational_profile),
  FUN = mean
)

corr_cols <- c(
  "autonomy_goal", "achievement_goal", "belonging_goal", "security_goal", "meaning_goal", "status_goal",
  "goal_conflict", "goal_ownership", "autonomy_support", "competence_support", "relatedness_support",
  "need_support", "motivational_quality", "life_direction_coherence", "conscientiousness",
  "persistence_score", "adaptive_disengagement", "well_being"
)

model_persistence <- lm(
  persistence_score ~ autonomy_goal + achievement_goal + belonging_goal +
    security_goal + meaning_goal + status_goal +
    goal_conflict + goal_ownership + conscientiousness,
  data = data
)

model_conflict_buffer <- lm(
  persistence_score ~ goal_conflict * conscientiousness +
    goal_ownership + motivational_quality + need_support,
  data = data
)

model_wellbeing <- lm(
  well_being ~ motivational_quality + life_direction_coherence +
    goal_conflict + adaptive_disengagement + conscientiousness,
  data = data
)

model_disengagement <- lm(
  adaptive_disengagement ~ goal_conflict + goal_ownership +
    motivational_quality + conscientiousness,
  data = data
)

write.csv(context_summary, file.path(outputs, "r_motivation_context_summary.csv"), row.names = FALSE)
write.csv(profile_summary, file.path(outputs, "r_motivation_profile_summary.csv"), row.names = FALSE)
write.csv(cor(data[, corr_cols], use = "pairwise.complete.obs"), file.path(outputs, "r_motivation_goals_correlations.csv"))
capture.output(summary(model_persistence), file = file.path(outputs, "r_persistence_model_summary.txt"))
capture.output(summary(model_conflict_buffer), file = file.path(outputs, "r_conflict_buffer_model_summary.txt"))
capture.output(summary(model_wellbeing), file = file.path(outputs, "r_wellbeing_model_summary.txt"))
capture.output(summary(model_disengagement), file = file.path(outputs, "r_adaptive_disengagement_model_summary.txt"))
write.csv(data, file.path(outputs, "r_scored_motivation_goals_desire.csv"), row.names = FALSE)
cat("Wrote R outputs to:", outputs, "\n")

File: articles/motivation-goals-and-the-architecture-of-desire/python/analyze_motivation_goals_desire.py
#!/usr/bin/env python3
from pathlib import Path
import json
import numpy as np
import pandas as pd

ROOT = Path(__file__).resolve().parents[1]
DATA = ROOT / "data" / "synthetic_motivation_goals_desire.csv"
OUT = ROOT / "outputs"
OUT.mkdir(exist_ok=True)

GOALS = [
    "autonomy_goal",
    "achievement_goal",
    "belonging_goal",
    "security_goal",
    "meaning_goal",
    "status_goal",
]
SUPPORT = ["autonomy_support", "competence_support", "relatedness_support"]
REGULATION = ["goal_conflict", "goal_ownership", "conscientiousness", "persistence_score", "adaptive_disengagement"]
DERIVED = [
    "total_goal_intensity",
    "approach_orientation",
    "avoidance_security_orientation",
    "status_orientation",
    "need_support",
    "motivational_quality",
    "life_direction_coherence",
]

def ols(df, y, xs):
    X = np.column_stack([np.ones(len(df)), df[xs].to_numpy(float)])
    beta, *_ = np.linalg.lstsq(X, df[y].to_numpy(float), rcond=None)
    return dict(zip(["intercept", *xs], map(float, beta)))

df = pd.read_csv(DATA)
required = {"person_id", "motivation_context", "well_being", *GOALS, *SUPPORT, *REGULATION, *DERIVED}
missing = required - set(df.columns)
if missing:
    raise ValueError(f"Missing columns: {sorted(missing)}")

df["ownership_level"] = np.where(df["goal_ownership"] > df["goal_ownership"].median(), "higher_ownership", "lower_ownership")
df["conflict_level"] = np.where(df["goal_conflict"] > df["goal_conflict"].median(), "higher_conflict", "lower_conflict")
df["meaning_level"] = np.where(df["meaning_goal"] > df["meaning_goal"].median(), "higher_meaning", "lower_meaning")
df["motivational_profile"] = df["ownership_level"] + "_" + df["conflict_level"] + "_" + df["meaning_level"]

df["high_conflict_low_ownership"] = (
    (df["goal_conflict"] > df["goal_conflict"].median())
    & (df["goal_ownership"] < df["goal_ownership"].median())
)
df["high_status_low_meaning"] = (
    (df["status_goal"] > df["status_goal"].median())
    & (df["meaning_goal"] < df["meaning_goal"].median())
)

context_summary = (
    df.groupby("motivation_context")
    .agg(
        n=("person_id", "count"),
        approach_orientation_mean=("approach_orientation", "mean"),
        status_orientation_mean=("status_orientation", "mean"),
        avoidance_security_orientation_mean=("avoidance_security_orientation", "mean"),
        need_support_mean=("need_support", "mean"),
        goal_ownership_mean=("goal_ownership", "mean"),
        motivational_quality_mean=("motivational_quality", "mean"),
        goal_conflict_mean=("goal_conflict", "mean"),
        persistence_score_mean=("persistence_score", "mean"),
        adaptive_disengagement_mean=("adaptive_disengagement", "mean"),
        life_direction_coherence_mean=("life_direction_coherence", "mean"),
        well_being_mean=("well_being", "mean"),
    )
    .reset_index()
)

profile_summary = (
    df.groupby("motivational_profile")
    .agg(
        n=("person_id", "count"),
        goal_ownership_mean=("goal_ownership", "mean"),
        goal_conflict_mean=("goal_conflict", "mean"),
        meaning_goal_mean=("meaning_goal", "mean"),
        need_support_mean=("need_support", "mean"),
        motivational_quality_mean=("motivational_quality", "mean"),
        life_direction_coherence_mean=("life_direction_coherence", "mean"),
        persistence_score_mean=("persistence_score", "mean"),
        adaptive_disengagement_mean=("adaptive_disengagement", "mean"),
        well_being_mean=("well_being", "mean"),
    )
    .reset_index()
)

pattern_summary = pd.DataFrame({
    "pattern": ["high_conflict_low_ownership", "high_status_low_meaning"],
    "n": [int(df["high_conflict_low_ownership"].sum()), int(df["high_status_low_meaning"].sum())],
    "proportion": [float(df["high_conflict_low_ownership"].mean()), float(df["high_status_low_meaning"].mean())],
})

models = {
    "persistence_score": ols(
        df,
        "persistence_score",
        [
            "autonomy_goal",
            "achievement_goal",
            "belonging_goal",
            "security_goal",
            "meaning_goal",
            "status_goal",
            "goal_conflict",
            "goal_ownership",
            "conscientiousness",
        ],
    ),
    "well_being": ols(
        df,
        "well_being",
        [
            "motivational_quality",
            "life_direction_coherence",
            "goal_conflict",
            "adaptive_disengagement",
            "conscientiousness",
        ],
    ),
    "adaptive_disengagement": ols(
        df,
        "adaptive_disengagement",
        ["goal_conflict", "goal_ownership", "motivational_quality", "conscientiousness"],
    ),
    "life_direction_coherence": ols(
        df,
        "life_direction_coherence",
        ["goal_ownership", "motivational_quality", "approach_orientation", "goal_conflict", "need_support"],
    ),
    "note": "Synthetic demonstration only; not moral ranking, clinical assessment, personality testing, workplace screening, legal evaluation, or individual prediction.",
}

corr_cols = GOALS + SUPPORT + REGULATION + DERIVED + ["well_being"]

context_summary.to_csv(OUT / "python_motivation_context_summary.csv", index=False)
profile_summary.to_csv(OUT / "python_motivation_profile_summary.csv", index=False)
pattern_summary.to_csv(OUT / "python_motivation_pattern_summary.csv", index=False)
df[corr_cols].corr().to_csv(OUT / "python_motivation_goals_correlations.csv")
(OUT / "python_motivation_goals_models.json").write_text(json.dumps(models, indent=2))
df.to_csv(OUT / "python_scored_motivation_goals_desire.csv", index=False)
print(f"Wrote outputs to {OUT}")
